Sources/SpeziViews/Views/ManagedNavigationStack/ManagedNavigationStack+Path.swift

```diff
@@ -0,0 +1,329 @@
+        path.append(identifier)
+    }
+    
+    /// Moves to the next navigation step.
+    ///
+    /// An invocation of this function moves the `Path` to the
+    /// next navigation step as outlined by the order of views within the ``ManagedNavigationStack``.
+    ///
+    /// The tracking of the current state of the navigation flow is done fully automatic by the ``ManagedNavigationStack/Path``.
+    ///
+    /// After all navigation steps have been shown, the injected `isComplete` binding is set to `true` indicating that the navigation flow is completed.
+    public func nextStep() {
+        guard let currentStepIndex = steps.elements.keys.firstIndex(where: { $0 == currentStep }),
+              currentStepIndex + 1 < steps.elements.count else {
+            isComplete?.wrappedValue = true
+            return
+        }
+        pushStep(identifiedBy: steps.elements.keys[currentStepIndex + 1])
+    }
+    
+    /// Modifies the navigation path to move to the next navigation step identified by the specified value, and also add all steps inbetween.
+    ///
+    /// This function will look at all navigation steps after the current step, identify the first one that matches `selector`, and push it onto the stack.
+    /// Optionally, it will also push all intermediate steps between the current step and the one matching `selector`.
+    ///
+    /// If no step matching `stepRef` exists, this function will have no effect.
+    ///
+    /// - parameter selector: The selector used to identify the first matching step.
+    /// - parameter includeIntermediateSteps: Whether the stack should include all intermediate steps between the current step and the one matching `id`.
+    ///
+    /// - Note: If `stepRef` is a `View` type, this function will navigate to the first step (following the current step) with a matching view type,
+    ///     even if that step is also using a custom identifier. If there are multiple steps with this type, and you want to select a step other than the first one,
+    ///     use an explicit identifier for matching instead.
+    public func navigateToNextStep(matching selector: StepSelector, includeIntermediateSteps: Bool) {
+        let currentIndex = currentStep.flatMap {
+            steps.keys.firstIndex(of: $0)
+        } ?? 0
+        guard let stepIdentifierIdx = steps.keys[currentIndex...].dropFirst().firstIndex(where: { stepIdentifier in
+            guard !stepIdentifier.isCustom else {
+                return false
+            }
+            switch (selector, stepIdentifier.identifierKind) {
+            case (.viewType(let type), _):
+                return stepIdentifier.viewType == type
+            case let (.identifier(valueA), .identifiable(valueB)):
+                return valueA.isEqual(valueB)
+            case (.identifier, .viewTypeAndSourceLoc):
+                return false
+            }
+        }) else {
+            // unable to find matching step; ignore navigation request.
```

Comment:
We had a similar behavior previously, correct?

Reply:
yes.